### ISSUE TITLE
getRefreshToken doesn't support async functions

### ISSUE THREAD

**Issue by Hampfh** [pyncz/trpc-refresh-token-link#3]
Cool library!

One problem I ran into trying to use it through was the function for extracting the refresh token which cannot be async which is problematic in scenarios where storage solutions return a promise. One example of this is AsyncStorage.getItem in react native. 

```ts
trpc.createClient({
  links: [
    refreshTokenLink({
      getRefreshToken: async () => {
        await AsyncStorage.getItem("refresh_token")
      }
    }),
    httpBatchLink({
      url: "http://localhost:3000/api/trpc"
    })
  ]
})
```

**Comment by pyncz** [comment]
Oh, yeah, makes sense. Thanks for the fix, Hampus!

**Comment by pyncz** [comment]
✅ Released in `v1.0.2`
